v1/src/test/java/com/google/cloud/teleport/bigtable/AvroToBigtablePerformanceIT.java

```diff
@@ -0,0 +1,165 @@
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Performance tests for {@link AvroToBigtable} (GCS_Avro_to_Cloud_Bigtable) template. */
+@TemplateIntegrationTest(AvroToBigtable.class)
+@RunWith(JUnit4.class)
+public class AvroToBigtablePerformanceIT extends PerformanceBenchmarkingBase {
+  private static final Logger LOG = LoggerFactory.getLogger(AvroToBigtablePerformanceIT.class);
+  private static final String SPEC_PATH =
+      MoreObjects.firstNonNull(
+          "gs://dataflow-templates/latest/GCS_Avro_to_Cloud_Bigtable", TestProperties.specPath());
+  private static final String ARTIFACT_BUCKET = TestProperties.artifactBucket();
+  private static final String TEST_ROOT_DIR =
+      AvroToBigtablePerformanceIT.class.getSimpleName().toLowerCase();
+  // 56,000,000 messages of the given schema approximately make up 10GB
+  private static final String NUM_MESSAGES = "56000000";
+  private static final String INPUT_PCOLLECTION = "Read from Avro/Read.out0";
+  private static final String OUTPUT_PCOLLECTION = "Transform to Bigtable.out0";
+  private static BigtableResourceManager bigtableResourceManager;
+  private static ArtifactClient artifactClient;
+  private static String generatorSchemaPath;
+  private static String avroSchemaPath;
+
+  @Before
+  public void setup() throws IOException {
+    // Set up resource managers
+    bigtableResourceManager =
+        DefaultBigtableResourceManager.builder(testName.getMethodName(), PROJECT).build();
+    Storage gcsClient = createGcsClient(CREDENTIALS);
+    artifactClient = GcsArtifactClient.builder(gcsClient, ARTIFACT_BUCKET, TEST_ROOT_DIR).build();
+    // upload schema files and save path
+    generatorSchemaPath =
+        getFullGcsPath(
+            ARTIFACT_BUCKET,
+            artifactClient
+                .uploadArtifact(
+                    "input/schema.json",
+                    Resources.getResource("AvroToBigtablePerformanceIT/schema.json").getPath())
+                .name());
+    avroSchemaPath =
+        getFullGcsPath(
+            ARTIFACT_BUCKET,
+            artifactClient
+                .uploadArtifact(
+                    "input/bigtable.avsc",
+                    Resources.getResource("schema/avro/bigtable.avsc").getPath())
+                .name());
+  }
+
+  @After
+  public void teardown() {
+    bigtableResourceManager.cleanupAll();
+    artifactClient.cleanupRun();
+  }
+
+  @Test
+  public void testBacklog10gb() throws IOException, ParseException, InterruptedException {
+    // Arrange
+    String name = testName.getMethodName();
+    String jobName = createJobName(name);
+    // Bigtable expects Avro files to meet a specific schema
+    DataGenerator dataGenerator =
+        DataGenerator.builderWithSchemaLocation(jobName + "-data-generator", generatorSchemaPath)
+            .setQPS("1000000")
```

Comment:
nit: how realistic is to have 1 million/s? Is it really the intention to cause backpressure on the DataGenerator? maybe having some constants (e.g. BACKPRESSURE_QPS, HEAVY_QPS) would help understanding

Reply:
1 million QPS is very unrealistic since we have a message limit of only 35 million. The intention is to cause backpressure on the data generator. The current implementation of Streaming data generator template creates stragglers since the fanout of `GenerateSequence.from(...).to(...)` is very high. This leads to costly and slow processing in the pipeline. I plan on fixing this by adding a `Reshuffle` before `Generate Fake Messages` step in the pipeline, but we are currently blocked by https://github.com/apache/beam/issues/25140.

Having a high QPS and a high number of starting workers reduces the fanout thereby improving processing time and cost (cost reduces by more than 100% for generating 10gb of data) 

Note: This issue doesn't occur for streaming pipelines from Streaming Data generator template, only for batch pipelines.

Reply:
Adding constants for clarity is a good idea. I'll do this in a separate PR since I will need to make changes to other performance tests too.